### PR TITLE
FW-5136 Add visibility parameter to the search API

### PR DESCRIPTION
**Pull request by NolanVH**
### Description of Changes
This PR adds a parameter `visibility` to the search API. The parameter adds filtering by document visibility: `Team`, `Members`, and `Public`. 

Note that the results will first be filtered by what the user has permission to view.

### Checklist
- ~README / documentation has been updated if needed~
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~
- [x] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/tests/test_search/test_filters.py [hunk 1]
```diff
@@ -207,3 +207,35 @@ def test_staff_permissions(self, role):
...
+    )
+    def test_team(self, visibility):
+        search_query = get_search_query(visibility=visibility)
+        search_query = search_query.to_dict()
+
+        filtered_terms = search_query["query"]["bool"]["filter"][0]["terms"]
+        assert "visibility" in filtered_terms
+
+        for value in visibility:
+            assert value in filtered_terms["visibility"]
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_search/test_filters.py:241]: Please also assert that visibility and filtered_terms["visibility"] have the same length. This test would currently pass if the query always included all visibility options.

File: firstvoices/backend/tests/test_search/test_query_builder_utils.py [hunk 2]
```diff
@@ -67,3 +69,30 @@ def test_invalid_input(self):
...
+        ],
+    )
+    def test_valid_inputs(self, input_visibility, expected_visibility):
+        actual_visibility = get_valid_visibility(input_visibility)
+        for value in actual_visibility:
+            assert value in expected_visibility
+
+    def test_invalid_input(self):
+        actual_visibility = get_valid_visibility("bananas")
+        assert actual_visibility is None
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_search/test_query_builder_utils.py:98]: 🍌🍌🍌

**Review by sarahfirstvoices — APPROVED** [review]
No need to re-review after that little test update

**Review by sahil97 — APPROVED** [review]
Works as expected on Local. +1 for Sarah's suggested change for the test.
LGTM !

**Pull request merged**